Src/DependencyCollector/Microsoft.AI.DependencyCollector/DependencyCollectorDiagnosticListener.cs

```
@@ -0,0 +1,235 @@
+        }
+
+        /// <summary>
+        /// Diagnostic event handler method for 'System.Net.Http.Request' event.
+        /// </summary>
+        [DiagnosticName("System.Net.Http.Request")]
+        public void OnRequest(HttpRequestMessage request, Guid loggingRequestId)
+        {
+            if (request != null && request.RequestUri != null && !this.applicationInsightsUrlFilter.IsApplicationInsightsUrl(request.RequestUri.ToString()))
+            {
+                string httpMethod = request.Method.Method;
+                Uri requestUri = request.RequestUri;
+                string resourceName = requestUri.AbsolutePath;
+                if (!string.IsNullOrEmpty(httpMethod))
+                {
+                    resourceName = httpMethod + " " + resourceName;
+                }
+
+                DependencyTelemetry telemetry = new DependencyTelemetry();
+                this.client.Initialize(telemetry);
+                telemetry.Start();
+                telemetry.Name = resourceName;
+                telemetry.Target = requestUri.Host;
+                telemetry.Type = RemoteDependencyConstants.HTTP;
+                telemetry.Data = requestUri.OriginalString;
+                this.pendingTelemetry.TryAdd(loggingRequestId, telemetry);
+
+                HttpRequestHeaders requestHeaders = request.Headers;
+                if (requestHeaders != null)
+                {
+                    if (!string.IsNullOrEmpty(telemetry.Context.InstrumentationKey) && !ContainsRequestContextKeyValue(requestHeaders, RequestResponseHeaders.RequestContextSourceKey))
+                    {
+                        string sourceApplicationId;
+                        if (this.correlationIdLookupHelper.TryGetXComponentCorrelationId(telemetry.Context.InstrumentationKey, out sourceApplicationId))
+                        {
+                            SetRequestContextKeyValue(requestHeaders, RequestResponseHeaders.RequestContextSourceKey, sourceApplicationId);
```

what is the exception handling story here. Getting or setting request context key values or other calls here might throw. We shouldn't let the exception bubble up to the user. If we failed to fetch and add the correlation header, we should still let the user's calls carry on.